cache/cache.go

```diff
@@ -1,77 +1,221 @@
 package cache
 
 import (
-	"io"
+	"encoding/json"
 	"os"
 	"path/filepath"
-	"strings"
 
+	"github.com/aquasecurity/fanal/types"
+
+	bolt "go.etcd.io/bbolt"
 	"golang.org/x/xerrors"
 )
 
 const (
 	cacheDirName = "fanal"
-)
 
-var (
-	replacer = strings.NewReplacer("/", "_")
+	// imageBucket stores image information with image ID
+	imageBucket = "image"
+	// layerBucket stores os, package and library information per layer ID
+	layerBucket = "layer"
+	// decompressedDigestBucket stores a mapping from any digest to an uncompressed digest.
+	decompressedDigestBucket = "decompressed"
 )
 
 type Cache interface {
-	Get(key string) (reader io.ReadCloser)
-	Set(key string, file io.Reader) (reader io.Reader, err error)
-	SetBytes(key string, value []byte) (err error)
+	ImageCache
+	LocalImageCache
```

Comment:
For my understanding: standalone and server can use `FsCache` which implements the full `Cache` interface and client uses `RemoteCache` which implements the `ImageCache` interface.

What's the purpose of `PutImage` and `GetImage`? 
- Another way to ask that question would be that can we re-use `PutLayer` and `GetLayer` to store the information in ImageInfo which is used in these methods. There's an overlap of some items like `SchemaVersion` and `OS` information.

Reply:
>For my understanding: standalone and server can use FsCache which implements the full Cache interface and client uses RemoteCache which implements the ImageCache interface.

Exactly. `RemoteCache` doesn't need `GetLayer`, `GetImage` and `Clear`, so I prepared the another interface.

Reply:
>Another way to ask that question would be that can we re-use PutLayer and GetLayer to store the information in ImageInfo which is used in these methods. There's an overlap of some items like SchemaVersion and OS information.

`PutImage` and `GetImage` is used to store image metadata in the cache. They are completely different from layer information. But I considered the same thing that we could replace `PutLayer` with `PutBlob` and reuse it for `ImageInfo` as well. But the `ImageInfo` has the different structure from `LayerInfo`. `OS` of `LayerInfo` is a result of analysis for each layer (e.g. `alpine` or `redhat`) and means OS of the image itself, but `OS` of `ImageInfo` is from Docker metadata, which indicates on which OS the image runs like `linux`, `macOS` or `windows. Actually, this is unnecessary for now. I store these metadata from Docker for the future.

If we reuse `PutBlob`, it is difficult to restrict a type. I mean the definition is like `PutBlob([]byte) or PutBlob(interface{})`. If we separate them, we can define `PutLayer(LayerInfo)` and `PutImage(ImageInfo)`. I'm not sure which is better.